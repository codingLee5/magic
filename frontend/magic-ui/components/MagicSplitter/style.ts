import { createStyles } from "antd-style"

export const useStyles = createStyles(({ css, prefixCls }) => {
	return {
		splitter: css`
			--${prefixCls}-splitter-split-bar-draggable-size: 0 !important;
			--${prefixCls}-splitter-split-trigger-size: 0 !important;
      

      .${prefixCls}-splitter-panel{
        padding: 0 !important;
        overflow: hidden;
      }

			.${prefixCls}-splitter-bar-dragger::before {
				background-color: transparent !important;

				&:hover {
					background-color: transparent !important;
				}
			}
		`,
	}
})
