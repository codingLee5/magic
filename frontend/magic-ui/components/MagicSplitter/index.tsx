import type { SplitterProps } from "antd"
import { Splitter } from "antd"
import type { PropsWithChildren } from "react"
import { memo } from "react"
import { useStyles } from "./style"

export type MagicSplitterProps = PropsWithChildren<SplitterProps>

const MagicSplitter = memo(function MagicSplitter({
	className,
	children,
	...props
}: MagicSplitterProps) {
	const { styles, cx } = useStyles()

	return (
		<Splitter className={cx(styles.splitter, className)} {...props}>
			{children}
		</Splitter>
	)
})

// @ts-ignore
MagicSplitter.Panel = Splitter.Panel

type CompoundedComponent = typeof MagicSplitter & {
	Panel: typeof Splitter.Panel
}

export default MagicSplitter as CompoundedComponent
