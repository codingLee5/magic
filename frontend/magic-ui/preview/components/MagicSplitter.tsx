import React from "react"
import MagicSplitter from "../../components/MagicSplitter"
import ComponentDemo from "./Container"

const SplitterDemo: React.FC = () => {
	return (
		<div>
			<ComponentDemo
				title="基础分割器"
				description="最基本的分割器组件"
				code="<MagicSplitter><div>左侧</div><div>右侧</div></MagicSplitter>"
			>
				<div style={{ border: "1px solid #d9d9d9" }}>
					<MagicSplitter>
						<MagicSplitter.Panel min={100}>
							<div style={{ padding: 20, background: "#f0f0f0" }}>左侧面板</div>
						</MagicSplitter.Panel>
						<MagicSplitter.Panel min={100}>
							<div style={{ padding: 20, background: "#e6f7ff" }}>右侧面板</div>
						</MagicSplitter.Panel>
					</MagicSplitter>
				</div>
			</ComponentDemo>

			<ComponentDemo
				title="垂直分割器"
				description="垂直方向的分割器"
				code="split='horizontal'"
			>
				<div style={{ border: "1px solid #d9d9d9" }}>
					<MagicSplitter layout="vertical" style={{ height: 300 }}>
						<MagicSplitter.Panel min={100}>
							<div style={{ padding: 20, background: "#f0f0f0", height: "100%" }}>
								上方面板
							</div>
						</MagicSplitter.Panel>
						<MagicSplitter.Panel min={100}>
							<div style={{ padding: 20, background: "#e6f7ff", height: "100%" }}>
								下方面板
							</div>
						</MagicSplitter.Panel>
					</MagicSplitter>
				</div>
			</ComponentDemo>

			<ComponentDemo title="组合分割器" description="组合分割器">
				<div style={{ border: "1px solid #d9d9d9" }}>
					<MagicSplitter>
						<MagicSplitter.Panel min={100}>
							<div style={{ padding: 20, background: "#f0f0f0", height: "100%" }}>
								左侧面板
							</div>
						</MagicSplitter.Panel>
						<MagicSplitter.Panel min={100}>
							<MagicSplitter layout="vertical" style={{ height: 300 }}>
								<MagicSplitter.Panel min={100}>
									<div
										style={{
											padding: 20,
											background: "pink",
											height: "100%",
										}}
									>
										上方面板
									</div>
								</MagicSplitter.Panel>
								<MagicSplitter.Panel min={100}>
									<div
										style={{
											padding: 20,
											background: "#e6f7ff",
											height: "100%",
										}}
									>
										下方面板
									</div>
								</MagicSplitter.Panel>
							</MagicSplitter>
						</MagicSplitter.Panel>
					</MagicSplitter>
				</div>
			</ComponentDemo>
		</div>
	)
}

export default SplitterDemo
