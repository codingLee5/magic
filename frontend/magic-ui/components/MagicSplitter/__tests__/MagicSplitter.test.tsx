import { render, screen } from "@testing-library/react"
import MagicThemeProvider from "../../ThemeProvider"
import { describe, it, expect } from "vitest"
import MagicSplitter from "../index"

const renderWithTheme = (component: React.ReactElement) =>
	render(<MagicThemeProvider theme="light">{component}</MagicThemeProvider>)

describe("MagicSplitter", () => {
	it("应该正常渲染", () => {
		renderWithTheme(
			<MagicSplitter>
				<div>左侧</div>
				<div>右侧</div>
			</MagicSplitter>,
		)
		expect(screen.getByText("左侧")).toBeInTheDocument()
		expect(screen.getByText("右侧")).toBeInTheDocument()
	})

	// 快照测试
	describe("快照测试", () => {
		it("基础分割器快照", () => {
			const { asFragment } = renderWithTheme(
				<MagicSplitter>
					<div>左侧内容</div>
					<div>右侧内容</div>
				</MagicSplitter>,
			)
			expect(asFragment()).toMatchSnapshot()
		})

		it("带自定义样式分割器快照", () => {
			const { asFragment } = renderWithTheme(
				<MagicSplitter style={{ height: "400px" }}>
					<div>左侧内容</div>
					<div>右侧内容</div>
				</MagicSplitter>,
			)
			expect(asFragment()).toMatchSnapshot()
		})

		it("带类名分割器快照", () => {
			const { asFragment } = renderWithTheme(
				<MagicSplitter className="custom-splitter">
					<div>左侧内容</div>
					<div>右侧内容</div>
				</MagicSplitter>,
			)
			expect(asFragment()).toMatchSnapshot()
		})

		it("带自定义样式分割器快照", () => {
			const { asFragment } = renderWithTheme(
				<MagicSplitter style={{ height: "400px" }}>
					<div>左侧内容</div>
					<div>右侧内容</div>
				</MagicSplitter>,
			)
			expect(asFragment()).toMatchSnapshot()
		})

		it("带类名分割器快照", () => {
			const { asFragment } = renderWithTheme(
				<MagicSplitter className="custom-splitter">
					<div>左侧内容</div>
					<div>右侧内容</div>
				</MagicSplitter>,
			)
			expect(asFragment()).toMatchSnapshot()
		})

		it("复杂内容分割器快照", () => {
			const { asFragment } = renderWithTheme(
				<MagicSplitter>
					<div>
						<h3>左侧标题</h3>
						<p>左侧详细内容</p>
					</div>
					<div>
						<h3>右侧标题</h3>
						<p>右侧详细内容</p>
					</div>
				</MagicSplitter>,
			)
			expect(asFragment()).toMatchSnapshot()
		})
	})
})
