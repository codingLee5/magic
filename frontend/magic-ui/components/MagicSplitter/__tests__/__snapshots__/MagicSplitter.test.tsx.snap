// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`MagicSplitter > 快照测试 > 基础分割器快照 1`] = `
<DocumentFragment>
  <div
    class="magic-splitter acss-1ms8eiu magic-splitter-horizontal css-var-r1 magic-splitter-css-var css-dev-only-do-not-override-yotw4n"
  >
    <div
      class="magic-splitter-panel"
      style="flex-basis: auto; flex-grow: 1;"
    >
      左侧内容
    </div>
    <div
      aria-valuemax="0"
      aria-valuemin="0"
      aria-valuenow="50"
      class="magic-splitter-bar"
      role="separator"
    >
      <div
        class="magic-splitter-bar-dragger"
      />
    </div>
    <div
      class="magic-splitter-panel"
      style="flex-basis: auto; flex-grow: 1;"
    >
      右侧内容
    </div>
  </div>
</DocumentFragment>
`;

exports[`MagicSplitter > 快照测试 > 复杂内容分割器快照 1`] = `
<DocumentFragment>
  <div
    class="magic-splitter acss-1ms8eiu magic-splitter-horizontal css-var-r6 magic-splitter-css-var css-dev-only-do-not-override-yotw4n"
  >
    <div
      class="magic-splitter-panel"
      style="flex-basis: auto; flex-grow: 1;"
    >
      <h3>
        左侧标题
      </h3>
      <p>
        左侧详细内容
      </p>
    </div>
    <div
      aria-valuemax="0"
      aria-valuemin="0"
      aria-valuenow="50"
      class="magic-splitter-bar"
      role="separator"
    >
      <div
        class="magic-splitter-bar-dragger"
      />
    </div>
    <div
      class="magic-splitter-panel"
      style="flex-basis: auto; flex-grow: 1;"
    >
      <h3>
        右侧标题
      </h3>
      <p>
        右侧详细内容
      </p>
    </div>
  </div>
</DocumentFragment>
`;

exports[`MagicSplitter > 快照测试 > 带类名分割器快照 1`] = `
<DocumentFragment>
  <div
    class="magic-splitter acss-1ms8eiu custom-splitter magic-splitter-horizontal css-var-r3 magic-splitter-css-var css-dev-only-do-not-override-yotw4n"
  >
    <div
      class="magic-splitter-panel"
      style="flex-basis: auto; flex-grow: 1;"
    >
      左侧内容
    </div>
    <div
      aria-valuemax="0"
      aria-valuemin="0"
      aria-valuenow="50"
      class="magic-splitter-bar"
      role="separator"
    >
      <div
        class="magic-splitter-bar-dragger"
      />
    </div>
    <div
      class="magic-splitter-panel"
      style="flex-basis: auto; flex-grow: 1;"
    >
      右侧内容
    </div>
  </div>
</DocumentFragment>
`;

exports[`MagicSplitter > 快照测试 > 带类名分割器快照 2`] = `
<DocumentFragment>
  <div
    class="magic-splitter acss-1ms8eiu custom-splitter magic-splitter-horizontal css-var-r5 magic-splitter-css-var css-dev-only-do-not-override-yotw4n"
  >
    <div
      class="magic-splitter-panel"
      style="flex-basis: auto; flex-grow: 1;"
    >
      左侧内容
    </div>
    <div
      aria-valuemax="0"
      aria-valuemin="0"
      aria-valuenow="50"
      class="magic-splitter-bar"
      role="separator"
    >
      <div
        class="magic-splitter-bar-dragger"
      />
    </div>
    <div
      class="magic-splitter-panel"
      style="flex-basis: auto; flex-grow: 1;"
    >
      右侧内容
    </div>
  </div>
</DocumentFragment>
`;

exports[`MagicSplitter > 快照测试 > 带自定义样式分割器快照 1`] = `
<DocumentFragment>
  <div
    class="magic-splitter acss-1ms8eiu magic-splitter-horizontal css-var-r2 magic-splitter-css-var css-dev-only-do-not-override-yotw4n"
    style="height: 400px;"
  >
    <div
      class="magic-splitter-panel"
      style="flex-basis: auto; flex-grow: 1;"
    >
      左侧内容
    </div>
    <div
      aria-valuemax="0"
      aria-valuemin="0"
      aria-valuenow="50"
      class="magic-splitter-bar"
      role="separator"
    >
      <div
        class="magic-splitter-bar-dragger"
      />
    </div>
    <div
      class="magic-splitter-panel"
      style="flex-basis: auto; flex-grow: 1;"
    >
      右侧内容
    </div>
  </div>
</DocumentFragment>
`;

exports[`MagicSplitter > 快照测试 > 带自定义样式分割器快照 2`] = `
<DocumentFragment>
  <div
    class="magic-splitter acss-1ms8eiu magic-splitter-horizontal css-var-r4 magic-splitter-css-var css-dev-only-do-not-override-yotw4n"
    style="height: 400px;"
  >
    <div
      class="magic-splitter-panel"
      style="flex-basis: auto; flex-grow: 1;"
    >
      左侧内容
    </div>
    <div
      aria-valuemax="0"
      aria-valuemin="0"
      aria-valuenow="50"
      class="magic-splitter-bar"
      role="separator"
    >
      <div
        class="magic-splitter-bar-dragger"
      />
    </div>
    <div
      class="magic-splitter-panel"
      style="flex-basis: auto; flex-grow: 1;"
    >
      右侧内容
    </div>
  </div>
</DocumentFragment>
`;
